module/modmanager/manager.go

```diff
@@ -581,10 +591,20 @@ func (m *module) startProcess(
 	if err := modlib.CheckSocketAddressLength(m.addr); err != nil {
 		return err
 	}
+
+	// Start module process with -debug if:
+	//  * debug is set to true on the module
+	//  * debug is unset on the module but manager log level is Debug
+	args := []string{m.addr}
+	if (m.debug != nil && *m.debug) ||
+		(m.debug == nil && logger.Level() == zapcore.DebugLevel) {
```

Comment:
```suggestion
		logger.LevelEnabled(zapcore.DebugLevel) {
```
I don't think it matters if m.debug is nil the second time, and LevelEnabled lets you check for any logging setting that includes debug (in case additional levels are ever added above debug.)

Reply:
> I don't think it matters if m.debug is nil the second time

Removing that piece of the conditional means that when `m.debug == &false` and the manager logger level is debug, we pass `-debug` to the module. I think we want to _not_ pass `-debug` when `m.debug == &false` no matter what the module manager's logger level is.

> and LevelEnabled lets you check for any logging setting

Could you point me to that method? It doesn't seem to be a method on `golog.Logger`.

Reply:
Sorry, that was my bad (writing suggestions in github instead of my editor) It's `logger.Level().Enabled(zapcore.DebugLevel)`

https://pkg.go.dev/go.uber.org/zap/zapcore#Level.Enabled


As to the other point... I thought the intent was to enable if EITHER was true.  I didn't realize that you were attempting to use a pointer to a bool as a sort of ternary option here.

With that in mind, I don't think a boolean pointer is the best idea, as protobuf is going to cause a boolean to be "false" by default when it's omitted ( https://protobuf.dev/programming-guides/proto3/#default ), so you'd end up having to do extra things in protobuf to ensure a true ternary value is encoded/decoded across the wire, and that's... fragile. (And other languages/SDKs may not even have an equivalent concept for a nil bool.)

If the three-way behavior is required, I'd suggest just making a string (or string-based enum) for "LogLevel" instead of a boolean for Debug, and then its easy to encode in protobuf and simply have the "default" inherit from the manager. Otherwise, users can specify "debug" or "info" (or whatever you want to define) to force what they want.

Bonus idea: might not be horrible to just let them pick any of the zapcore levels for full control.

Reply:
Thanks! Used `logger.Level().Enabled...`.

After a bit of offline discussion, it seems:
-  we'd like to keep the ability to set a log level and only inherit from the server if unset 
-  we'd like to use a string setting for now to be most extensible for future log levels (let me know if you all think an enum would be better) 
- we'd like to use a double-dashed command line argument like `--log-level=debug`

The `NewLoggerFromArgs` utility is now _super_ basic, and just checks if `--log-level=debug` is the third argument in `os.Args` (I could switch back to actually parsing flags if that's desirable, but we're back to having to handle an error sans logger).